### ISSUE TITLE
Long job logs breaking job rotation by the operator

### ISSUE THREAD

**Issue by pcheliniy** [aquasecurity/starboard#287]
**What steps did you take and what happened:**
We had to build 0.8.x operator version from the current mainline to have a possibility to make vuln scans for private repo images.
Everything works fine, but sometimes jobs stuck and aren't removed by operator.
After some time 3 jobs stuck and it prevents new scans.

**What did you expect to happen:**
Jobs don't stuck and correctly removed by operator and scans of new workloads work correctly.

**Anything else you would like to add:**
We saw the following problem in operator logs:
```
020-12-21T09:51:06.070Z        DEBUG   controller.job  Resolving workload properties   {"job": "starboard-operator/caf73314-0c0c-4840-b248-25e178674c6f", "owner": {"Kind":"ReplicaSet","Name":"talent-app-production-sidekiq-84d466fbf7","Namespace":"talent"}, "hash": "8c8696d54", "containerImages": {"talent-app-production-sidekiq":"reg.talenttechlab.org/talent/app:v1.0.21"}}
2020-12-21T09:51:06.096Z       ERROR   controller      Reconciler error        {"reconcilerGroup": "batch", "reconcilerKind": "Job", "controller": "job", "name": "caf73314-0c0c-4840-b248-25e178674c6f", "namespace": "starboard-operator", "error": "invalid character '}' looking for beginning of value"}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:246
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:218
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:197
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.Until
        /go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:90
```

We decided to look at the job logs and the log was a little bit cropped. [job-result.json.txt](https://github.com/aquasecurity/starboard/files/5724012/job-result.json.txt)
Our assumption is that sometimes the log is too long, and this causes problems.
Maybe it depends on k8s configuration or something like this but we have no idea how to fix it.
Would be appreciate if you offer us a workaround solution.

**Environment:**

- Starboard version (use `starboard version`):   
   git rev-parse --short HEAD - 02ffc30
- Kubernetes version (use `kubectl version`):
   Client Version:  v1.18.12
   Server Version: v1.18.9

**Comment by danielpacak** [comment]
Hi @pcheliniy Thanks for reporting this problem. I'll try to figure out what is going on. In the meantime, are you able to tell which container images are usually failing and whether I can pull them from public registry such as Quay.io or DockerHub? This might help me with reproduction.

**Comment by pcheliniy** [comment]
Unfortunately, it's impossible to provide absolutely the same image.
But I made quite a similar image based on the same base image + apt packages + the same gems collection.
It's available on dockerhub `pcheliniy/longlogs:0.0.1`

I've tried to reproduce the mentioned problem with this image and fortunately, I did it.

**Comment by danielpacak** [comment]
@pcheliniy I tried scanning the `pcheliniy/longlogs:0.0.1` with Starboard CLI and it actually worked in my dev environment. Although it has plenty vulnerabilities and huge logs output (~8.1M), the vulnerability report was saved in etcd successfully.

![longlogs](https://user-images.githubusercontent.com/1322923/103007178-3f257480-4533-11eb-9219-12f0fbb98b46.png)

Also the logs output is not cropped:

```
$ head longlog.txt
[
  {
    "Target": "pcheliniy/longlogs:0.0.1 (debian 9.12)",
    "Type": "debian",
    "Vulnerabilities": [
      {
        "VulnerabilityID": "CVE-2020-27350",
        "PkgName": "apt",
        "InstalledVersion": "1.4.9",
        "FixedVersion": "1.4.11",
$ tail longlog.txt
          "https://nvd.nist.gov/vuln/detail/CVE-2020-15169",
          "https://weblog.rubyonrails.org/2020/9/10/Rails-5-2-4-4-and-6-0-3-3-have-been-released",
          "https://www.debian.org/security/2020/dsa-4766"
        ],
        "PublishedDate": "2020-09-11T16:15:00Z",
        "LastModifiedDate": "2020-12-08T18:58:00Z"
      }
    ]
  }
]
```

I'm wondering if you can use Starboard CLI and do the same test in your env and confirm that it's still also failing?
And to confirm that it's related to the logs size, could you adjust `trivy.severity: HIGHT,CRITICAL` and see if it helps?

**Comment by pcheliniy** [comment]
@danielpacak I think we've figured out the problem.
K8s have some limitations in logging, I mean it rotates logs every 10Mb https://kubernetes.io/docs/concepts/cluster-administration/logging/#logging-at-the-node-level (see note at the end of the pointed paragraph)
This "rotation size" can be configured for kubelet and our cloud provider has a little bit harder settings :)

The problem appears if only we use full scan with all security levels. Frankly speaking, having information about low priority vulnerabilities is not so crucial for us. That's why your solution not to use such verbose settings quite reasonable.

Thanks for the help, and I guess we can close the issue.

**Comment by danielpacak** [comment]
I'm glad that you sorted it out and I'm going to close the issue. If the logs rotation is still an issue we can reopen it and thing about solutions, but I hope you only run workloads with low number of vulnerabilities ;)
